src/main/java/atmosfears/AtmosFearsBE/service/AirParticulatesService.java

```diff
@@ -1,31 +1,65 @@
 package atmosfears.AtmosFearsBE.service;
 
-import atmosfears.AtmosFearsBE.database.AirParticulatesFileRepository;
 import atmosfears.AtmosFearsBE.database.AirParticulatesRepository;
+import atmosfears.AtmosFearsBE.database.Particulate;
 import atmosfears.AtmosFearsBE.model.AirParticulates;
-import atmosfears.AtmosFearsBE.model.AirParticulatesFile;
 import lombok.AllArgsConstructor;
+import lombok.RequiredArgsConstructor;
+import org.json.JSONObject;
 import org.springframework.stereotype.Service;
 
-import java.util.List;
+import java.util.*;
 
 @Service
 @AllArgsConstructor
 public class AirParticulatesService {
 
-    private final AirParticulatesRepository airParticulatesRepository;
+    @RequiredArgsConstructor
+    static class AverageParticulateCounter {
+        public final Particulate name;
+        private int occurrences = 0;
+        private double valuesSum = 0;
+
+        void addValue(Double value) {
+            if (value != null) {
+                valuesSum += value;
+                occurrences++;
+            }
+        }
+
+        double getValue() {
+            return occurrences == 0 ? 0 : valuesSum / occurrences;
+        }
+    }
 
-    private final AirParticulatesFileRepository airParticulatesFileRepository;
```

Comment:
In next PR's, with change in `DataIngestController`, this Repisitory should be deleted pernamently